packages/expo/tsconfig.base.json

```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "lib": ["dom", "esnext"],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "noEmit": true,
+    "allowSyntheticDefaultImports": true,
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "moduleResolution": "node",
+    "jsx": "react-native"
```

Comment:
Some of these options like "target: esnext" and "jsx: react-native" seem mandatory while others like "skipLibCheck" seem like they might be optional. Ideally we'd move the more optional stuff out so that our base tsconfig focuses more on what is truly needed or super common.

Reply:
These are required for keeping projects working in TS as closely to JS as possible, this is optimal for users migrating to TypeScript from a JavaScript project. Overtime they can add more strictness if needed. `forceConsistentCasingInFileNames` is not _required_ so we can remove it. `skipLibCheck` is a recommended feature and keeps the compilation fast. 
```
    "target": "esnext",
    "lib": ["dom", "esnext"],
    "allowJs": true,
    "noEmit": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "moduleResolution": "node",
    "jsx": "react-native"
```

Reply:
For a base config, we should lean towards minimalism and not making people who have their ducks in a row have to turn off rules that don't apply to them because they've done a good job. In this case I think it's just "allowJs" we might not want to include. As you said, it's for transitions and full TS projects don't need it (I think?).

Reply:
I think two things that make `allowJs` good are:
- The TSConfig is kinda accounting for Metro bundler by default, which is why we don’t transform JSX or emit JS files. Likewise, Metro will still work with JS files, so `typecheck` should probably continue to work. Metro also doesn’t consider JS files to be “assets” but rather as application code, if allowJs isn’t present then TS kinda treats JS like assets by not bundling them with the code.
- Migration from a base , and things like RNWebView account for JS files.